test/test_nn.py

```diff
@@ -16208,9 +16208,10 @@ def test_pooling_bfloat16(self, device):
         self._test_bfloat16_ops(torch.nn.AdaptiveAvgPool2d((3, 5)), device, inp_dims=(8, 4, 16, 16), prec=0.05)
         self._test_bfloat16_ops(torch.nn.AdaptiveAvgPool3d((3, 5, 7)), device, inp_dims=(8, 4, 16, 16, 16), prec=0.05)
 
-    @onlyCUDA
+    @onlyOnCPUAndCUDA
     def test_softmax_bfloat16(self, device):
-        self._test_bfloat16_ops(torch.nn.Softmax(dim=1), device, inp_dims=(16, 32), prec=1e-2)
+        for dim in [0, 1]:
```

Comment:
Is it useful to broaden this to more dimensions and larger inputs?

Reply:
Sure, broaden the test to 4 dim and test the tail part with special input shapes. As for the accuracy bound, looks like most of the BF16 test case are using [precision 1e-2](https://github.com/pytorch/pytorch/blob/a9c6aa291bbd18f2236e20434610fd3c672215ba/test/test_nn.py#L16193-L16200). Since we have lots of bf16<->fp32 cast in softmax, it will lose some accuracy. I believe it's hard to tighten the accuracy bound.